### ISSUE TITLE
[Events] Implement NCA, VBA, and Vet Center facilities as Event locations

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#10536]
## Status
**[2024-05-17]** The following facilities should be available in Drupal via the Facilities API and should be able to be part of this implementation. Daniel to update the estimate (next week).
**[2024-05-07]** NCA's exist in nodes in Drupal, so discovery is not needed and the work can commence.

## Description

Editors need the ability to choose any VA Facility as an Event Location when they're creating or editing events. The following event locations should now be available in Drupal via the Facilities API and should be made available to editors as events locations:
- NCA
- VBA
- Vet Center
- Vet Center Outstation

_Note that VAMC was removed from the list because VAMC Facilities are already available as event locations (encapsulates Medical Centers and VA Clinics). We want to ensure as part of this implementation not to regress the existing VAMCs (but there's nothing additional from VAMC content types that needs to be added right now)._ 

~From 8/30 Wes/Jill sync with Dave: 
National Cemetery Association (m.va.gov) facilities need to migrate full data to Drupal. Currently, the content type exists but only lists operating status & geolocation: 
https://prod.cms.va.gov/admin/content?title=&type=nca_facility&moderation_state=All&owner=All
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10535 tracks an effort to get NCAs listed as Event locations. In order for that to be possible, we need address data available to display in front-end Event templates. 
This ticket represents the discovery to work out what's required to flesh out NCA nodes, include address, and enable #10535. Facilities folks have performed / are performing similar work for VBA facilities, so @dsinla and Facilities Drupal folks or Swirt are a good place to start with conversations to understand the work.~

## Acceptance Criteria
- [ ] NCA Facilities appear and be selectable in the event location dropdown for all users (user should be able to specify a cemetery)
- [ ] VAMC Facilities appear and be selectable in the event location dropdown for all users (user should be able to specify a VAMC facility)
- [ ] VBA Facilities appear and be selectable in the event location dropdown for all users (user should be able to specify a VBA facility)
- [ ] Vet Center Facilities appear and be selectable in the event location dropdown for all users (user should be able to specify a Vet Center facility)
- [ ] Vet Center Outstation Facilities appear and be selectable in the event location dropdown for all users (user should be able to specify a Vet Center Outstation facility)
- [ ] Existing VAMCs (Medical Centers and VA Clinics) are still available to be selected
- [ ] Create an event located at at least one of each of the facilities above to check the Front-end and create a ticket for FE work if needed.

**Comment by swirtSJW** [comment]
Basic outline of steps
* audit data available in facility api for NCAs
* Decide what should be migrated
* Map data to exisitng fields from facilities, vet centers and VBA
* If there is data that does not map to existing fields, fields should be spec'd out.
* Update existing nca migration to add the additional fields.

**Comment by dsinla** [comment]
While performing review of tickets in backlog:
Discussing with @JayDarnell:
Entity reference field on Events does not currently support NCA facility nodes as a possible target 

The question is around priority - @dsinla to discuss where this falls in long term roadmap with @davidconlon 
Are there services at NCA facilities? Do we just expand on data to support events? then come back later for services?

**Comment by FranECross** [comment]
Description updated to include all facilities migrated into Drupal via the facilities API (NCA, VAMC, VBA, Vet Center, Vet Center Outstation). Another ticket will be created to accommodate any facilities not yet in Drupal to determine how to make them available to the CMS front end for editors to choose.

**Comment by dsasser** [comment]
@FranECross I'm a bit unclear on which 'VAMC' facilities should be allowed. Should we allow any of these that begin with VAMC?

![Screenshot 2024-05-20 at 4 42 41 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/16d35a4d-5037-4b43-ad21-7aa7c1d5eb6d)

**Comment by FranECross** [comment]
@dsasser Regarding your  question above☝🏼... I see what you mean. Those aren't locations, they're pages/info/etc. Let me get clarification from Dave C, unless @jilladams knows offhand if there's a way to narrow by locations that events might me at?

**Comment by jilladams** [comment]
Noted in Slack: VAMC Facilities (which covers medical centers and VA clinics) are already exposed as locations that can be chosen on Events. So I think the note here is just not to regress that, but there's nothing additional from VAMC content types that needs to be added right now. (Or prob ever.)

**Comment by FranECross** [comment]
Thanks, @jilladams ! I tweaked the Description a bit, and also added AC to ensure the existing VAMC Medical Centers and VA Clinics are still available.

**Comment by jilladams** [comment]
@dsasser BEFORE YOU MERGE: I think per Dave, the vision here is that a VAMC or Outreach editor should be able to create an Event using any Facility as the location. But: I would like to test a couple things (or: make sure that a couple of things are tested)  in a Tugboat for my own state of mind, before we ship: 
* As a Vet Center or VBA editor: 
  * Am I able to create Events? 
  * In my section? Or in what sections? (I just don't know if Vet Center or VBA editors have Event creation perms.) 
  * If so, and they publish, what breadcrumb do they use in the FE?  (Vet Center & VBA pages don't have a sidebar or Event Listing for Events yet)

**Comment by dsasser** [comment]
@jilladams 

> As a Vet Center or VBA editor...Am I able to create Events?

No. Only the 'Content creator - Outreach Hub', 'Content creator - VAMC', and 'Content admin' roles can _create_ Events, but other roles can edit them after the fact providing the Events fall within their assigned Section.

A Tugboat is available [here](https://tugboat.vfs.va.gov/665a293a17e38363d36a4e77) for testing.

**Comment by jilladams** [comment]
Verified: 
* Vet Center content creator cannot make events
* As VAMC editor, created an event at a Vet Center (and on Natl outreach cal)
  * Nothing changed there: can still only select my own VAMC section and Event listing
  * Event: https://cms-l9y1tokkkuehk5yz646oouagmvwv4dqe.ci.cms.va.gov/node/69472/edit 
* Same is true for Outreach Editor: only Outreach section / listing.(didn't save one of these) 

I requested content release to build the FE & verify, but: it'll just be a reg VAMC / Outreach event, and location won't affect crumbs. 

I am reassured. I'll take a look at the FE tomorrow to make sure, but when your PR is ready, should be ok to merge once that's out of the way.
